x-pack/plugins/security_solution/public/common/components/events_tab/histogram_configurations.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import numeral from '@elastic/numeral';
+
+import { MatrixHistogramType } from '../../../../common/search_strategy/security_solution';
+import { getExternalAlertLensAttributes } from '../visualization_actions/lens_attributes/common/external_alert';
+import { getEventsHistogramLensAttributes } from '../visualization_actions/lens_attributes/hosts/events';
+import type { MatrixHistogramConfigs, MatrixHistogramOption } from '../matrix_histogram/types';
+import * as i18n from './translations';
+
+const DEFAULT_EVENTS_STACK_BY = 'event.action';
+
+export const getSubtitleFunction =
+  (defaultNumberFormat: string, isAlert: boolean) => (totalCount: number) =>
+    `${i18n.SHOWING}: ${numeral(totalCount).format(defaultNumberFormat)} ${
+      isAlert ? i18n.ALERTS_UNIT(totalCount) : i18n.EVENTS_UNIT(totalCount)
+    }`;
+
+export const eventsStackByOptions: MatrixHistogramOption[] = [
+  {
+    text: 'event.action',
+    value: 'event.action',
+  },
+  {
+    text: 'event.dataset',
+    value: 'event.dataset',
+  },
+  {
+    text: 'event.module',
+    value: 'event.module',
+  },
+];
+
+export const eventsHistogramConfig: MatrixHistogramConfigs = {
+  defaultStackByOption:
+    eventsStackByOptions.find((o) => o.text === DEFAULT_EVENTS_STACK_BY) ?? eventsStackByOptions[0],
+  errorMessage: i18n.ERROR_FETCHING_EVENTS_DATA,
+  histogramType: MatrixHistogramType.events,
+  stackByOptions: eventsStackByOptions,
+  subtitle: undefined,
+  title: i18n.EVENTS_GRAPH_TITLE,
+  getLensAttributes: getEventsHistogramLensAttributes,
+};
+
+export const alertsStackByOptions: MatrixHistogramOption[] = [
+  {
+    text: 'event.category',
+    value: 'event.category',
+  },
+  {
+    text: 'event.module',
+    value: 'event.module',
+  },
+];
+
+const DEFAULT_STACK_BY = 'event.module';
+
+export const alertsHistogramConfig: MatrixHistogramConfigs = {
+  defaultStackByOption:
+    alertsStackByOptions.find((o) => o.text === DEFAULT_STACK_BY) ?? alertsStackByOptions[1],
```

Comment:
It looks like the fallback `alertsStackByOptions[1]` (instead of `alertsStackByOptions[0]`) in the expression:

```ts
alertsStackByOptions.find((o) => o.text === DEFAULT_STACK_BY) ?? alertsStackByOptions[1]
```

may be an intended workaround for a bug where, in `8.3.2`, the `External alert trend` `Stack by` defaults to `event.module`:

![8_3_2_stack_by_default](https://user-images.githubusercontent.com/4459398/180512918-1c44d877-ae8e-4155-bfce-1b12751341f2.png)

but in the PR branch, it's currently defaulting to `event.category`:

![pr_branch_stack_by_default](https://user-images.githubusercontent.com/4459398/180513739-9f527bed-2d81-4d11-bb44-6b4f21650850.png)

The workaround isn't working as-intended:

- The `find` clause is not hitting the fallback case; it's (correctly) finding the `DEFAULT_STACK_BY`, which is `event.module`
- In the `8.3.2` release, there were two separate instances of the `MatrixHistogram`  component: One on the `Host > Events` tab, and another on the `Hosts > External alerts` tab.

The reason the workaround isn't working in the PR branch is the `MatrixHistogram` component only selects the default `Stack by` option when the component is first rendered. In the PR branch, when the user selects the `Show only external alerts` checkbox, the same instance of the `MatrixHistogram` component is being (re)used.

The `MatrixHistogram` component currently only selects the default at initialization time, so the re-used instance isn't selecting the new default when its props are updated.

As a result, the current default of the `External alerts trend` is `event.category`, instead of the expected `event.module`

## Suggested updates

- Update the current expression from:

```ts
alertsStackByOptions.find((o) => o.text === DEFAULT_STACK_BY) ?? alertsStackByOptions[1]
```
to

```ts
alertsStackByOptions.find((o) => o.text === DEFAULT_STACK_BY) ?? alertsStackByOptions[0]
```

- Update the `MatrixHistogram` component in `x-pack/plugins/security_solution/public/common/components/matrix_histogram/index.tsx` to have the following `useEffect`, which selects the default when an updated `defaultStackByOption` is provided:

```ts
  useEffect(() => {
    setSelectedStackByOption(defaultStackByOption);
  }, [defaultStackByOption]);
```

- Add a unit test that asserts `event.module` is the default `Stack by` for the `External alert trend` when users check the `Show only external alerts` option